src/compiler/sys.ts

```diff
@@ -1379,10 +1388,12 @@ namespace ts {
                             directories.push(entry);
                         }
                     }
+                    files.sort();
+                    directories.sort();
                     return { files, directories };
                 }
                 catch (e) {
-                    return emptyFileSystemEntries;
+                    return { files: [], directories: [] };

```

Comment:
Why did this change?  Merge issues?

Reply:
No, this is required to make your claim that “`getAccessibleFileSystemEntries` returns a new array for directories” true.

Reply:
@sheetalkamat is much better equipped to comment than I am, but maybe `getDirectories` should just return a readonly array?

Reply:
I looked through the code and in all cases we use that as readonly string[] .. But changing this would be breaking change as you would need to update many other hosts (module resolution host etc). I think we should separate that check from this... 
This can still be `readonly` if `getDirectories` is implemented as :
```ts
const result = getFileSystemAccessibleEntries();
return result == emptyFileSystemEntries ? [] : result.directories as string[];
```

Reply:
> `result.directories as string[]`

Unless you direct otherwise, I’d prefer not to insert a type-unsafe cast justified by non-local reasoning just so we can micro-optimize an _exception handler_.

Reply:
While I generally agree with the sentiment that exceptions should be reserved for exceptional circumstances, my observation has been that that is not how node uses them, at least for file system access.  I can't tell, just by looking at the code, how often that catch block is hit.  However, since someone saw fit to create a constant, I assume it was a problem at some point.

Having said that, I have no objection to merging this as-is.  If it turns out to be a problem, it's not hard to fix later.

Reply:
`git blame` shows the constant was created in #21244 while fixing an unrelated bug in `core.ts` and was used here just because it could be. Before that this was `{ files: [], directories: [] }`.

Reply:
I agree that that looks like a speculative optimization, but I'd like to get @sheetalkamat's sign-off before merging.

Reply:
I dont think we should change type of `getAccessibleFileSystemEntries` as part of this. Keeping that readonly is ideal since except getDirectory() which is a new code path everywhere else its readonly array.. I am not sure how much impact that will have to convert it to non readonly but i wouldnt want to combine that here.

Reply:
Alright, I’ve removed that commit (so the `.slice()` in `getDirectory` is back).